cmd/label_list.go

```diff
@@ -56,25 +56,29 @@ lab label list remote "search term"  # search "remote" for labels with "search t
 }
 
 func MapLabels(rn string, labelTerms []string) ([]string, error) {
+	var ambiguous bool
+
 	labels, err := lab.LabelList(rn)
 	if err != nil {
 		return nil, err
 	}
 
 	matches := make([]string, len(labelTerms))
 	for i, term := range labelTerms {
+		ambiguous = false
 		lowerTerm := strings.ToLower(term)
 		for _, label := range labels {
 			lowerLabel := strings.ToLower(label.Name)
 
-			// no match, or we already have an exact match
-			if !strings.Contains(lowerLabel, lowerTerm) || matches[i] == lowerLabel {
+			// no match
+			if !strings.Contains(lowerLabel, lowerTerm) {
 				continue
 			}
 
 			// only allow ambiguity for exact matches
 			if matches[i] != "" && lowerTerm != lowerLabel {
-				return nil, errors.Errorf("Label '%s' is ambiguous", term)
+				ambiguous = true
+				continue
 			}
 
 			matches[i] = label.Name
```

Comment:
Maybe do something like
```
    ambiguous = false;
    if lowerTerm == lowerLabel {
        break;
    }
```
here? That would set `ambiguous` if there are more than one substring matches, but no exact match (which was the intended behavior all along, but I failed to consider multiple substring matches before finding an exact match)

Reply:
I was thinking a bit more about the `ambiguous` state here: is it even worth informing the user if we're keeping exact match only behavior?

Reply:
> is it even worth informing the user if we're keeping exact match only behavior?

If there is an exact match, then it's not useful in my opinion.

But I'd still like to allow non-exact matches when a term can be matched to a single label (for selfish reasons, because gitlab.gnome.org's labels aren't very CLI friendly).

Reply:
>But I'd still like to allow non-exact matches when a term can be matched to a single label (for selfish reasons, because gitlab.gnome.org's labels aren't very CLI friendly).

ahh ok! I remember you sharing the link to them in some PR :).
I'm going to rework this patch.

Thanks for all the feedback @fmuellner

Reply:
Should be fixed in the new patch version

Reply:
Thanks!

Looks good to me at a quick glance.